code/modules/mob/living/silicon/robot/robot_mob.dm

```diff
@@ -1223,19 +1224,16 @@ GLOBAL_LIST_INIT(robot_verbs_default, list(
 		cell = null
 	qdel(src)
 
-#define BORG_CAMERA_BUFFER 3 SECONDS
+#define CAMERA_UPDATE_COOLDOWN 2.5 SECONDS
 
-/mob/living/silicon/robot/Move(atom/newloc, direct, movetime)
-	var/oldLoc = loc
+/mob/living/silicon/robot/Moved(atom/OldLoc, Dir, Forced)
 	. = ..()
-	if(. && !updating && camera)
-		updating = TRUE
-		spawn(BORG_CAMERA_BUFFER)
-			if(camera && oldLoc != loc)
-				GLOB.cameranet.updatePortableCamera(camera)
-			updating = FALSE
-
-#undef BORG_CAMERA_BUFFER
+	if(last_camera_update + CAMERA_UPDATE_COOLDOWN < world.time)
+		last_camera_update = world.time
+		GLOB.cameranet.updatePortableCamera(camera, OldLoc)
+		SEND_SIGNAL(camera, COMSIG_CAMERA_MOVED, OldLoc)
```

Comment:
camera is not guaranteed to be null here, causing signal runtimes

```
[2023-04-22T13:00:14] Runtime in robot_mob.dm,1236: Cannot read null.comp_lookup
[2023-04-22T13:00:14]   proc name: Moved (/mob/living/silicon/robot/Moved)
[2023-04-22T13:00:14]   src: the maintenance drone (187) (/mob/living/silicon/robot/drone)
[2023-04-22T13:00:14]   src.loc: the floor (177,138,2) (/turf/simulated/floor/plasteel)
[2023-04-22T13:00:14]   call stack:
[2023-04-22T13:00:14]   the maintenance drone (187) (/mob/living/silicon/robot/drone): Moved
[2023-04-22T13:00:14]   the maintenance drone (187) (/mob/living/silicon/robot/drone): forceMove
[2023-04-22T13:00:14]   the maintenance drone (187) (/mob/living/silicon/robot/drone): forceMove
[2023-04-22T13:00:14]   the maintenance drone (187) (/obj/item/holder/drone): process
[2023-04-22T13:00:14]   Objects (/datum/controller/subsystem/processing/obj): fire
[2023-04-22T13:00:14]   Objects (/datum/controller/subsystem/processing/obj): ignite
[2023-04-22T13:00:14]   Master (/datum/controller/master): RunQueue
[2023-04-22T13:00:14]   Master (/datum/controller/master): Loop
[2023-04-22T13:00:14]   Master (/datum/controller/master): StartProcessing
```

Reply:
Yep thanks! The new camera check should catch this